cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/dataset/output/MockInputFormat.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.batch.dataset.output;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Noop inputformat that generates mock input splits and asserts correct configuration values.
+ */
+public class MockInputFormat extends InputFormat<LongWritable, Text> {
+
+  /**
+   * Get noop splits
+   */
+  @Override
+  public List<InputSplit> getSplits(JobContext jobContext) throws IOException, InterruptedException {
+    Configuration configuration = jobContext.getConfiguration();
+    assert configuration.get("additionalProp").equals("source");
```

Comment:
The key should be a constant somewhere (maybe the app?) so that it's not hardcoded in 3 different classes.

Reply:
don't use asserts to test conditions in tests. They are not really used anywhere and are more meant to check conditions that must never happen while having an easy flag to turn off the checks. You should throw an exception instead.